packages/checkout/registry/index.js

```diff
@@ -0,0 +1,64 @@
+let checkoutFilters = {};
+
+/**
+ * Register filters for a specific extension.
+ *
+ * @param {string} namespace Name of the extension namespace.
+ * @param {Object} filters   Object of filters for that namespace. Each key of
+ *                           the object is the name of a filter.
+ */
+export const __experimentalRegisterCheckoutFilters = ( namespace, filters ) => {
+	checkoutFilters = {
+		...checkoutFilters,
+		[ namespace ]: filters,
+	};
+};
+
+/**
+ * Get all filters with a specific name.
+ *
+ * @param {string} filterName   Name of the filter to search for.
+ * @return {Function[]} Array of functions that are registered for that filter
+ *                      name.
+ */
+const getCheckoutFilters = ( filterName ) => {
+	const namespaces = Object.keys( checkoutFilters );
+	const filters = namespaces
+		.map( ( namespace ) => checkoutFilters[ namespace ][ filterName ] )
+		.filter( Boolean );
+	return filters;
+};
+
+/**
+ * Apply a filter.
+ *
+ * @param {Object}   o              Object of arguments.
+ * @param {string}   o.filterName   Name of the filter to apply.
+ * @param {any}      o.defaultValue Default value to filter.
+ * @param {any}      [o.arg]        Argument to pass to registered functions. If
+ *                                  several arguments need to be passed, use an
+ *                                  object.
+ * @param {Function} [o.validation] Function that needs to return true when the
+ *                                  filtered value is passed in order for the
+ *                                  filter to be applied.
+ * @return {any} Filtered value.
+ */
+export const __experimentalApplyCheckoutFilter = ( {
+	filterName,
+	defaultValue,
+	arg = null,
+	validation = () => true,
+} ) => {
+	const filters = getCheckoutFilters( filterName );
+	let value = defaultValue;
+	filters.forEach( ( filter ) => {
+		try {
+			const newValue = filter( value, arg );
+			value = validation( newValue ) ? newValue : value;
+		} catch ( e ) {
+			// eslint-disable-next-line no-console
+			console.log( e );
```

Comment:
Logging errors is not very practical, it wouldn't trigger an error on the console (easily missed) and would happen to all users or admins.
I will write a P2 to suggest how we can catch and expose errors, for now, how about you check if the current use is an admin, fi so, you throw, or else you neglect the error and move on.
```suggestion
			if ( CURRENT_USER_IS_ADMIN ) {
				throw e;
			}
```

Reply:
We will probably need to revisit this after the P2 discussion, but for now I made the change you suggested (throw when the user is an admin) but still kept the `console.error` for costumers: 171276669b2a55dc0f6cc170cd4304d8c6681d4b.